io/driver/module_unload_load.py

```diff
@@ -0,0 +1,144 @@
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def get_depend_modules(self, module):
+        """
+        Returns the dependent modules
+        """
+        config_path = os.path.join(os.path.abspath(''),
+                                   "module_unload_load.py.data/config")
+        for line in genio.read_all_lines(config_path):
+            if module in line:
+                return line.split('=')[-1]
+
+    def flush_mpath(self, mdl):
+        """
+        flush the multipath
+        """
+        cmd = "lsmod | grep -i ^%s" % mdl
+        for i in range(20):
+            if process.getoutput(cmd).split(" ")[-1] == '0':
+                return True
+            process.system("multipath -F", ignore_status=True)
+            time.sleep(3)
+            continue
+        return False
+
+    def module_load_unload(self, module_list):
+        """
+        Unloading and loading the given module
+        """
+        for mod1 in module_list:
+            if linux_modules.module_is_loaded(mod1) is False:
+                linux_modules.load_module(mod1)
+                time.sleep(self.load_unload_sleep_time)
+
+        for mdl in module_list:
+            for _ in range(0, self.iteration):
+                sub_mod = self.get_depend_modules(mdl)
+                if sub_mod:
+                    for mod in sub_mod.split(' '):
+                        if mod == 'multipath':
+                            if self.flush_mpath(mdl) is False:
+                                self.error_modules.append(mdl)
+                                break
+                        else:
+                            self.log.info("unloading sub module %s " % mod)
+                            linux_modules.unload_module(mod)
+                            if linux_modules.module_is_loaded(mod) is True:
+                                self.error_modules.append(mod)
+                                break
+                self.log.info("unloading module %s " % mdl)
+                linux_modules.unload_module(mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is True:
+                    self.error_modules.append(mdl)
+                    break
+                self.log.info("loading module : %s " % mdl)
+                linux_modules.load_module(mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is False:
+                    self.error_modules.append(mdl)
+                    break
+
+    def test(self):
+        """
+        Begining the test here
+        """
+        pci_addrs = []
+        if self.module:
+            for modl in self.module.split(','):
+                self.mod_list.append(modl)
+        elif self.only_io is True:
+            for pci_addrs in pci.get_pci_addresses():
+                self.mod_list.append(pci.get_driver(pci_addrs))
```

Comment:
this is not duplicate, the adapter is 4 port adapter hence with io_only True, it will check for all the pci_adress which is 4 pci_adress here. hence the list has 4 times thesame driver.

Reply:
@Naresh-ibm I understand that, but why do you want to run the test 4x times for qla2xxx, where x is the iteration specified by the user in yaml

Reply:
As I mentioned you earlier, it takes from PCI_ADRESS, if the adapter has two ports it takes it two times and so on. if you want me to take care of duplicates, I can do that.
As you mentioned previously that while porting we should not change the functionality of existing test case, I kept it as it is.

Reply:
@Naresh-ibm Right. Want this as an enhancement. I am fine with adding this as a new commit also.